### PR TITLE
Set metadata_proxy_shared_secret from secret

### PR DESCRIPTION
**Pull request by mrKisaoLamb**
Make the metadata_proxy_shared_secret configurable and modify the nova.conf template to set the secret only for nova-metadata. This change will require a follow-up modification in the install_yamls file to set both secrets for nova-metadata and ovn-metadata https://github.com/openstack-k8s-operators/install_yamls/pull/211.

### CODE REVIEW

File: test/functional/base_test.go [hunk 1]
```diff
@@ -66,7 +66,8 @@ func GetDefaultNovaAPISpec() map[string]interface{} {
 		"apiMessageBusSecretName": MessageBusSecretName,
 		"cell0DatabaseHostname":   "nova-cell0-db-hostname",
 		"keystoneAuthURL":         "keystone-auth-url",
-		"containerImage":          ContainerImage,
+
```
Comment (gibizer) [inline test/functional/base_test.go:69]: unnecessary whitespace?

File: config/crd/bases/nova.openstack.org_novaconductors.yaml [hunk 2]
```diff
@@ -154,6 +154,11 @@ spec:
                     description: CellDatabase - the name of the field to get the Cell
                       DB password from the Secret
                     type: string
+                  metadataSecret:
+                    default: NovaMetadataSecret
+                    description: MetadataSecret - the name of the field to get the
+                      metadata secret from the Secret
+                    type: string
```
[the remark this answers is not in the document]
Comment (mrKisaoLamb) [reply config/crd/bases/nova.openstack.org_novaconductors.yaml]: Thanks! done b22e6e2c72584e2cc2cb8407e6441d54e1aea05c

File: controllers/novaapi_controller.go [hunk 3]
```diff
@@ -362,7 +363,7 @@ func (r *NovaAPIReconciler) generateConfigs(
 		"default_project_domain": "Default",   // fixme
 		"default_user_domain":    "Default",   // fixme
 		"transport_url":          string(apiMessageBusSecret.Data["transport_url"]),
-		"metadata_secret":        "42", // fixme
+		"metadata_secret":        string(secret.Data[instance.Spec.PasswordSelectors.MetadataSecret]),
```
[the remark this answers is not in the document]
Comment (mrKisaoLamb) [reply controllers/novaapi_controller.go]: Thanks! done b22e6e2c72584e2cc2cb8407e6441d54e1aea05c

File: test/functional/base_test.go (hunk 1 above)
Comment (mrKisaoLamb) [reply test/functional/base_test.go]: Thanks! done b22e6e2c72584e2cc2cb8407e6441d54e1aea05c

File: test/functional/base_test.go [hunk 4]
```diff
@@ -636,6 +639,7 @@ func CreateNovaMetadataSecret(namespace string, name string) *corev1.Secret {
 			"NovaPassword":              []byte("12345678"),
 			"NovaAPIDatabasePassword":   []byte("12345678"),
 			"NovaCell0DatabasePassword": []byte("12345678"),
+			"NovaMetadataSecret":        []byte("12345678"),
```
[the remark this answers is not in the document]
Comment (mrKisaoLamb) [reply test/functional/base_test.go]: Thanks! done b22e6e2c72584e2cc2cb8407e6441d54e1aea05c

**Review by gibizer — COMMENTED** [review]
Looks good to me. Once we settle on the install_yamls PR I will deploy these two together to verify the integration. Then we can land the install_yamls PR first, then we can land this PR. We need this order otherwise devs using install_yamls will get a nova-operator that fails due to the install_yamls generated Secret does not have metadata field.

**Comment by gibizer** [comment]
/hold
Until https://github.com/openstack-k8s-operators/install_yamls/pull/211 lands.

**Review by gibizer — APPROVED** [review]

**Comment by gibizer** [comment]
/test nova-operator-build-deploy
https://github.com/openstack-k8s-operators/install_yamls/pull/211 landed

**Comment by gibizer** [comment]
/unhold
dependency landed in install_yamls, and it is effective as the e2e deploy job here is green again.

**Review by gibizer — APPROVED** [review]
